statsmodels/tsa/vector_ar/var_model.py

```diff
@@ -802,22 +832,95 @@ def is_stable(self, verbose=False):
         """
         return is_stable(self.coefs, verbose=verbose)
 
-    def plotsim(self, steps=1000):
+    def simulate_var(self, steps=None, offset=None, seed=None):
+        """
+        simulate the VAR(p) process for the desired number of steps
+
+        Parameters
+        ----------
+        steps : None or int
+            number of observations to simulate, this includes the initial
+            observations to start the autoregressive process.
+            If offset is not None, then exog of the model are used if they were
+            provided in the model
+        offset : None or ndarray (steps, neqs)
```

Comment:
I'm not sure I understand what this represents?  It looks a little bit like optional realizations for shocks(?)

Reply:
Not for shocks directly, those are still added with random draws.
The term offset is in analogy to GLM.